torch/utils/_sympy/solve.py

```diff
@@ -115,7 +115,10 @@ def _try_isolate_lhs(
 
         # If we can't tell whether 'other' is negative or positive, we do nothing.
         # That is because we don't know whether we have mirror the operation or not.
-        if not (isinstance(e, INEQUALITY_TYPES) and other.is_negative is None):
+        # We also divide only when we know 'rhs' is not zero.
+        if not (isinstance(e, INEQUALITY_TYPES) and other.is_negative is None) and not (
+            not isinstance(e, INEQUALITY_TYPES) and rhs.is_zero
```

Comment:
Should we instead avoid this if we do not definitely know that rhs is NOT zero? Similar to how the `is_negative` test is an is None test.